barter-data/src/exchange/bybit/book/l1.rs

```diff
@@ -0,0 +1,141 @@
+use crate::{
+    event::{MarketEvent, MarketIter},
+    exchange::{bybit::channel::BybitChannel, subscription::ExchangeSub, ExchangeId},
+    subscription::book::{Level, OrderBookL1},
+    Identifier,
+};
+use barter_integration::model::{Exchange, SubscriptionId};
+use chrono::{DateTime, Utc};
+use serde::{Deserialize, Serialize};
+
+#[derive(Debug, Deserialize)]
+pub struct BybitOrderBookL1 {
+    pub topic: String,
+    #[serde(with = "chrono::serde::ts_milliseconds")]
+    pub ts: DateTime<Utc>,
+    #[serde(rename = "type")]
+    pub update_type: String,
+    pub data: BybitOrderBookL1Data,
+    pub cts: u64,
+}
+
+#[derive(Debug, Deserialize)]
+pub struct BybitOrderBookL1Data {
+    pub s: String,
+    pub b: Vec<[String; 2]>,
+    pub a: Vec<[String; 2]>,
```

Comment:
since its fixed size don't use a Vec:

```suggestion
    pub b: [[f64; 2]],
    pub a: [[f64; 2]],
```
or 
```rust
    pub b: [(f64, f64)],
    pub a: [(f64, f64)],
```

Reply:
Since it's an L1 book and we don't use that outer layer of the data can we just do [f64; 2] or is there any benefit to it being nested like that

Reply:
the nesting is because I think bybit serializes it as a nested list, not sure if it can work without the nesting

Reply:
Sorry i might not be getting this like if you look at the current commit it looks like 
```rust
    #[serde(deserialize_with = "de_ob_l1_level")]
    pub b: [f64; 2], // [price, amount]
    #[serde(deserialize_with = "de_ob_l1_level")]
    pub a: [f64; 2], // [price, amount]
```

works rn since the `de_ob_l1_level` function returns a  `Result<[f64; 2], D::Error>` - cause only the implementation `de_ob_l1_level` needs to know the real bybit response format right? or does the struct have to mirror the bybit response somehow

Reply:
actually I misread this whole thing haha, yeah don't need the nesting. aslong as the deser code does the un-nesting I mean